src/main/java/org/hibernate/models/internal/IsResolvedTypeSwitch.java

```diff
@@ -31,17 +33,12 @@
 public class IsResolvedTypeSwitch implements TypeDetailsSwitch<Boolean> {
 	public static final IsResolvedTypeSwitch IS_RESOLVED_SWITCH = new IsResolvedTypeSwitch();
 
-	public static boolean isBound(TypeDetails typeDetails, SourceModelBuildingContext buildingContext) {
-		return TypeDetailsSwitch.switchType( typeDetails, IS_RESOLVED_SWITCH, buildingContext );
+	private static boolean isBound(TypeDetails typeDetails, SourceModelBuildingContext buildingContext) {
+		return TypeDetailsSwitch.switchType( typeDetails, IS_BOUND_SWITCH, buildingContext );
 	}
 
 	@Override
 	public Boolean caseClass(ClassTypeDetails classType, SourceModelBuildingContext buildingContext) {
-		// not completely kosher, but works for our needs
-		//noinspection RedundantIfStatement
-		if ( Objects.equals( classType.getClassDetails(), ClassDetails.OBJECT_CLASS_DETAILS ) ) {
-			return false;
-		}
```

Comment:
This is fine, but we could also just override `isResolved()` with a default method in the `ClassTypeDetails` interface that always returns `true`, so we won't have the need of another type switch.

Reply:
In the meantime, going to apply this.